### DISCUSSION TITLE
Debugging using Pix

### DISCUSSION THREAD

**Discussion by Gavin-Williams** [amerkoleci/Vortice.Windows#263] [Q&A]
I can't attack my process to Pix, I get the following error msg..

![image](https://user-images.githubusercontent.com/4374176/165465071-fe936df3-e60f-4618-a3da-c77bf5b8092f.png)

What's going on?

**Comment by amerkoleci** [comment]
Vortice uses legacy pix.h support and not pix3.h support, probably need to add support for it.
Let me think

**Reply by Gavin-Williams** [reply]
OK. I'll use that in the mean time. Another thing I noticed is that VS graphics diagnostics doesn't work. I know it's not really supported. But if it can work, it's a quick and handy tool.

**Reply by Axiverse** [reply]
What version do you have? I am using PIX 2201.24 without any issues.

**Reply by Gavin-Williams** [reply]
I have 2203.30

I can launch my WinUI app from Pix (as UWP) but the graphics command queue looks like..
0 Signal
1 Signal
2 Present

And the screenshot shows black except for some xaml text.

**Reply by Gavin-Williams** [reply]
And, something else, if I try to use VS to capture the frame I get ...

> Unhandled exception at 0x00007FFE59CA2082 (Microsoft.ui.xaml.dll) in DemoTests.exe: 0xC000027B: An application-internal exception has occurred (parameters: 0x000001ABA2BA6730, 0x0000000000000002).

**Reply by Axiverse** [reply]
Hmmm maybe a UWP issue. I just installed the latest 2203.30 and it works fine with Win32 .net 6 app. Something could also be misconfigured in the device, I feel like I've had it not work when I set up something wrong in the past. Code that you can share?

**Reply by Gavin-Williams** [reply]
This is a WindowsAppSDK / WinUI app. I was launching it via the UWP path. If I try the Win32 path with...

Path to exe = "F:\Projects\EngineProjects\DemoTriangle\bin\x64\Debug\net6.0-windows10.0.19041.0\win10-x64\AppX\DemoTests.exe"

Working directory = F:\Projects\EngineProjects ( I also tried .. EngineProjects\DemoTriangle) .. I honestly don't know what a working directory is.

I get failed to capture application - the title exited before performing any GPU work.

**Reply by amerkoleci** [reply]
Thats probably issue with WinUI stack, probably it renders using d3d11

**Reply by Gavin-Williams** [reply]
I'm using D3d11 as well, and I have Force D3D11on12 selected.

**Comment by Gavin-Williams** [comment]
@amerkoleci Is the legacy version available on this page?
https://devblogs.microsoft.com/pix/download/

**Comment by Gavin-Williams** [comment]
Just following up on this. I cannot find the old Pix at all. And I'm told it doesn't work after Windows 7. I've also tried using RenderDoc but I can't get it to launch the app. And VS Gfx Diagnostics doesn't work either.
